### PR TITLE
feat: provider config support insecureMode

### PR DESCRIPTION
**Pull request by sfc-gh-swinkler**
Allow support for optional insecureMode parameter to bypass OCSP checks

### CODE REVIEW

File: pkg/provider/provider.go [hunk 1]
```diff
@@ -164,6 +164,12 @@ func Provider() *schema.Provider {
 				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_PROTOCOL", "https"),
 			},
+			"insecure_mode": {
+				Type:        schema.TypeBool,
+				Description: "If true, bypass the Online Certificate Status Protocol (OCSP) certificate revocation check. IMPORTANT: Change the default value for testing or emergency situations only..",
```
Comment (sfc-gh-ngaberel) [inline pkg/provider/provider.go:169]:
```suggestion
				Description: "If true, bypass the Online Certificate Status Protocol (OCSP) certificate revocation check. IMPORTANT: Change the default value for testing or emergency situations only.",
```

File: docs/data-sources/shares.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (sfc-gh-ngaberel) [inline docs/data-sources/shares.md:1]: Should this be in a different PR?

Comment (sfc-gh-swinkler) [reply docs/data-sources/shares.md:1]: this is just the result of "make docs". so it means current main is not up to date.

**Review by sfc-gh-ngaberel — APPROVED** [review]
